libfreerdp/core/redirection.c

```diff
@@ -207,7 +207,7 @@ int rdp_redirection_apply_settings(rdpRdp* rdp)
 			settings->TargetNetAddresses[i] = _strdup(redirection->TargetNetAddresses[i]);
 			if (!settings->TargetNetAddresses[i])
 			{
-				for (--i; i >= 0; --i)
+				for (; i > 0; --i)
```

Comment:
Good catch - however I think now we miss out on freeing settings->TargetNetAddresses[0] - I'd propose following solution:

```
UINT32 j;
for (j = 0; j < i; ++j)
    free(settings->TargetNetAddresses[j]);
```